### PR TITLE
chore(Datalist): Allow reset to empty in restricted mode

### PR DESCRIPTION
**Pull request by tlnd-mhuchet**
**What is the problem this PR is trying to solve?**
When used in `restricted` mode, the datalist cannot be reset to an empty value, unless that empty value is explicitly an option in the `titleMap`.

**What is the chosen solution to this problem?**
Allow the reset of the value in restricted mode by simply cleaning the input, and persist the change (i.e. call the onChange props).

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

**Review by romainseb — APPROVED** [review]

File: packages/components/src/Datalist/Datalist.component.js [hunk 1]
```diff
@@ -78,6 +78,10 @@ function findEntry(titleMap, attributeName, attributeValue = '') {
 }
 
 function getEntryFromName(titleMap, name, restricted) {
+	if (name === '') {
+		return { name: '', value: '' };
```
Comment (Nigui) [inline packages/components/src/Datalist/Datalist.component.js:82]: you could return name directly

**Review by jsomsanith-tlnd — APPROVED** [review]

**Review by Nigui — APPROVED** [review]

File: packages/components/src/Datalist/Datalist.component.js (hunk 1 above)
Comment (jsomsanith-tlnd) [reply packages/components/src/Datalist/Datalist.component.js:82]: The format of the titlemap entries is `{name, value}`

Comment (Nigui) [reply packages/components/src/Datalist/Datalist.component.js:82]: sorry, I meant `{name,value:''}`

Comment (tlnd-mhuchet) [reply packages/components/src/Datalist/Datalist.component.js:82]:
```suggestion
		return { name, value: '' };
```

**Pull request merged**